src/parse.d

```diff
@@ -6985,6 +6991,9 @@ public:
             auto e2 = parseAssignExp();
             e = new CommaExp(loc, e, e2);
             loc = token.loc;
+            // Only allowed within `for` loop
+            if (!allow_commaexp)
+                deprecation("Comma expressions are deprecated");
```

Comment:
why is this in the loop? doesn't this yield the deprecation message more often then?
Btw please add a final EOL and deprecation date ;-)